### PR TITLE
Bash Code Review Recipe

### PR DESCRIPTION
**Pull request by lakshaykaushik**
Code Review Recipe for bash with formatting, linting, build pipeline, pre-commit and popular vs code extension changes

### CODE REVIEW

**Comment by lakshaykaushik** [comment]
@p-schuler , can you please review this PR.

**Comment by p-schuler** [comment]
Could we follow the same template we use in other recipes? For example: https://github.com/microsoft/code-with-engineering-playbook/blob/master/code-reviews/recipes/CSharp.md

**Comment by lakshaykaushik** [comment]
Made it consistent as per the template , do we want to retain the Dependancy section?

File: code-reviews/recipes/Bash.md [hunk 1]
```diff
@@ -22,6 +64,57 @@ To automate this process in Azure Devops you can add the following snippet to yo
...
+    rev: v0.7.1.1
+    hooks:
+    -   id: shellcheck
+ ```
+
+### Step 3
+
+Run `$ pre-commit install` to setup the git hook scripts
 
 ## Dependencies
```
Comment (p-schuler) [inline code-reviews/recipes/Bash.md:119]: The Code Review Checklist is still empty. Something we can add here?

Comment (lakshaykaushik) [reply code-reviews/recipes/Bash.md:119]: Done.

File: code-reviews/recipes/Bash.md [hunk 2]
```diff
@@ -41,4 +134,9 @@ fi
 
 In addition to the [Code Review Checklist](../README.md) you should also look for these bash specific code review items
 
-> TODO: Add checklist
+* [ ] Does this code use [Built-in Shell](https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html) Options like set -o, set -e, set -u for execution control of shell scripts.
```
Comment (ChrisRisner) [inline code-reviews/recipes/Bash.md:137]: Nit: This should end in a question mark.

File: code-reviews/recipes/Bash.md [hunk 3]
```diff
@@ -41,4 +134,9 @@ fi
...
+* [ ] Is the code modularized? Shell scripts can be modularized like python modules. Portions of bash scripts should be sourced in complex bash projects.
+* [ ] Are all exceptions handled correctly? Exceptions should be handled correctly using exit codes or trapping signals.
+* [ ] Does the code pass all linting checks as per shellcheck and unit tests as per shunit2.
```
Comment (ChrisRisner) [inline code-reviews/recipes/Bash.md:140]: Nit: This should end in a question mark.

File: code-reviews/recipes/Bash.md [hunk 4]
```diff
@@ -41,4 +134,9 @@ fi
...
+* [ ] Does the code uses relative paths or absolute paths? Relative paths should be avoided as they are prone to environment attacks. If relative path is needed, check the ```PATH``` variable is set.
```
Comment (ChrisRisner) [inline code-reviews/recipes/Bash.md:141]: Nits: "Does the code **use**".  ", check **that** the ```PATH``` "

File: code-reviews/recipes/Bash.md [hunk 5]
```diff
@@ -41,4 +134,9 @@ fi
...
+* [ ] Does the code takes credentials as user input? Are the credentials masked or encrypted in the script?
```
Comment (ChrisRisner) [inline code-reviews/recipes/Bash.md:142]: Nit: "Does the code **take**"

File: code-reviews/recipes/Bash.md (hunk 4 above)
Comment (lakshaykaushik) [reply code-reviews/recipes/Bash.md]: Done

File: code-reviews/recipes/Bash.md (hunk 5 above)
Comment (lakshaykaushik) [reply code-reviews/recipes/Bash.md]: Done

File: code-reviews/recipes/Bash.md (hunk 3 above)
Comment (lakshaykaushik) [reply code-reviews/recipes/Bash.md]: Done

File: code-reviews/recipes/Bash.md (hunk 2 above)
Comment (lakshaykaushik) [reply code-reviews/recipes/Bash.md]: Done

Comment (lakshaykaushik) [reply code-reviews/recipes/Bash.md]: @ChrisRisner , made changes, can you please review and approve the PR.

**Pull request merged**